### ISSUE TITLE
bug: XDSText hasTruncateTooltip never triggers for multi-line truncation (maxLines > 1)

### ISSUE THREAD

**Issue by bobenheimer** [facebook/astryx#1710]
## Bug

`XDSText` with `maxLines >= 2` and `hasTruncateTooltip` never shows the truncation tooltip, even when text is visually clamped with an ellipsis.

Single-line truncation (`maxLines={1}`) works correctly.

## Root Cause

In `useTruncation.ts` (line 84), multi-line truncation detection uses:

```ts
setIsTruncated(element.scrollHeight > element.offsetHeight);
```

When `-webkit-line-clamp` is active (via `display: -webkit-box` + `overflow: hidden` + `-webkit-box-orient: vertical`), Chrome may clamp `scrollHeight` to equal `offsetHeight`. This is because the `-webkit-line-clamp` implementation fragments content away rather than simply hiding overflow — the browser reports the clamped box dimensions as the total content dimensions.

Single-line (`maxLines === 1`) works correctly because it uses a different code path:

```ts
setIsTruncated(element.scrollWidth > element.offsetWidth);
```

Single-line truncation uses `text-overflow: ellipsis` + `white-space: nowrap` instead of `-webkit-line-clamp`, so `scrollWidth` reliably reports the full content width.

## Reproduction

```tsx
<XDSText type="body" maxLines={2}>
  This is a very long text that should definitely overflow beyond two lines
  when rendered in a narrow container. The tooltip should appear on hover
  but it never does because isTruncated is always false.
</XDSText>
```

Expected: hovering over the clamped text shows a tooltip with the full text.
Actual: no tooltip appears. `isTruncated` remains `false`.

## Suggested Fix

Replace `scrollHeight` comparison with `Range.getBoundingClientRect().height`, which reports the actual content height regardless of `-webkit-line-clamp`:

```ts
// In checkTruncation, the multi-line branch:

// Before (broken):
setIsTruncated(element.scrollHeight > element.offsetHeight);

// After (working):
const range = document.createRange();
range.selectNodeContents(element);
const contentHeight = range.getBoundingClientRect().height;
range.detach();
setIsTruncated(contentHeight > element.offsetHeight);
```

`Range.getBoundingClientRect()` measures the bounding box of the actual text content nodes, bypassing the clamped box layout entirely. This technique is well-established for detecting overflow in `-webkit-line-clamp` scenarios.

## Affected Files

- `packages/core/src/Text/useTruncation.ts` — line 84
- `packages/core/src/Text/text.stylex.ts` — `truncationStyles.multiLine` (for context)

## Notes

- The `scrollHeight` behavior with `-webkit-line-clamp` may vary across browser versions — the fix using `Range` is more robust across all browsers.
- The `range.detach()` call is a no-op in modern browsers but included for completeness.
